### PR TITLE
add types for 'bagpipes'

### PR DESCRIPTION
**Pull request by micmro**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/bagpipes/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    /** Dynamic values passed to the fitting during execution */
+    input?: any;
+    /** The name of the context key to which the output value is assigned */
+    output?: any;
+    // allow other config settings
+    [prop: string]: any;
+}
+
+/** Middleware style `next` callback  */
+type Callback = (err: Error | null | undefined, res?: any) => void;
```
Comment (andy-ms) [inline types/bagpipes/index.d.ts]: This is only used in one place; inline it and provide a type for `res`.

File: types/bagpipes/tslint.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "extends": "../tslint.json" }
```
Comment (andy-ms) [inline types/bagpipes/tslint.json]: We've recently updated it; please merge in master and change the value to `dtslint/dt.json`.

File: types/bagpipes/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    /** add a new `Fitting` (piece) to the pipe (aka. pipeline) */
+    fit(pipe: Fitting): Pipe;
+    /** add a new `Fitting` (piece) to the pipe (aka. pipeline) */
+    fit(options: PipeworksOptions, pipe: Fitting): Pipe;
+    /** Redirect the flow to another pipe (aka. pipeline). */
+    siphon(pipe: Fitting): Pipe;
+    /** Redirect the flow to another pipe (aka. pipeline). */
+    siphon(options: PipeworksOptions, pipe: Fitting): Pipe;
+    /**
+     * Send something down the pipe (aka. pipeline)! Any number of arguments can be sent,
```
Comment (andy-ms) [inline types/bagpipes/index.d.ts]: Any number of arguments can be sent: so `...args: any[]`?

File: types/bagpipes/index.d.ts (hunk 1 above)
Comment (micmro) [inline types/bagpipes/index.d.ts]: I've inlined it but `res` is based on external inputs (from `input`), so it could be anything.

**Comment by micmro** [comment]
Thanks @andy-ms, I fixed the issues highlighted.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**